### PR TITLE
Fix ProductSliderCmsElementResolver criteria association

### LINKED ISSUE #4757: Dynamic product groups manufacturer are no longer displayed in the dynamic product groups after the update 6.6.6.0
*PHP Version:* 8.2   | *Shopware Version:* 6.6.6.0  | *Affected area / extension:* Extension:Shape-Theme 

 ----

*_Expected behaviour:_*

Dynamic product groups manufacturer are no longer displayed in the dynamic product groups after the update 6.6.6.0

*_Actual behaviour:_*

Dynamic product groups manufacturer are no longer displayed in the dynamic product groups after the update 6.6.6.0

*_How to reproduce:_*

Dynamic product groups manufacturer are no longer displayed in the dynamic product groups after the update 6.6.6.0

### LINKED ISSUE #5137: after the update to 6.6.6.1 the manufacturers are no longer displayed in the product slider from the dynamic product groups
*PHP Version:* 8,2   | *Shopware Version:* 6.6.6.1   | *Affected area / extension:* Platform(Default) 

 ----

*_Expected behaviour:_*

**### after the update to 6.6.6.1 the manufacturers are no longer displayed in the product slider from the dynamic product groups**

*_Actual behaviour:_*

**### after the update to 6.6.6.1 the manufacturers are no longer displayed in the product slider from the dynamic product groups**

*_How to reproduce:_*

**### after the update to 6.6.6.1 the manufacturers are no longer displayed in the product slider from #the dynamic product groups**

### PR DESCRIPTION
**Pull request by ghost**
### 1. Why is this change necessary?
- If you use the product slider cms element and configure it to use a product stream as the source for products, the `manufacturer` association is missing and no product manufacturers are displayed in the frontend

### 2. What does this change do, exactly?
* Changed `Content/Product/Cms/ProductSliderCmsElementResolver.php` to include the missing `manufacturer` association for product-stream products

### 3. Describe each step to reproduce the issue or behaviour.
- Use the product-slider cms element on a cms page
- Configure it to use a product stream as the source for products
- Open the product slider in the storefront to see, that there is no manufacturer data

### 4. Please link to the relevant issues (if any).
- Fixes #4757
- Fixes #5137

### 5. Checklist

- [X] I have rebased my changes to remove merge conflicts
- [X] I have written tests and verified that they fail without my change
- [X] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [X] I have written or adjusted the documentation according to my changes
- [X] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfill them.

### CODE REVIEW

File: src/Core/Content/Product/Cms/ProductSliderCmsElementResolver.php [hunk 1]
```diff
@@ -59,7 +59,6 @@ public function collect(CmsSlotEntity $slot, ResolverContext $resolverContext):
 
         if ($products->isStatic() && $products->getValue()) {
             $criteria = new Criteria($products->getArrayValue());
-            $criteria->addAssociation('cover');
```
Comment (mitelg) [inline src/Core/Content/Product/Cms/ProductSliderCmsElementResolver.php:62]: why did you remove the cover? :thinking:

Comment (ghost) [reply src/Core/Content/Product/Cms/ProductSliderCmsElementResolver.php:62]: @mitelg Because we don't need it here. It's already present by https://github.com/shopware/shopware/blob/trunk/src/Core/Content/Product/SalesChannel/SalesChannelProductDefinition.php#L60 and will be overriden by this anyway.

Comment (mitelg) [reply src/Core/Content/Product/Cms/ProductSliderCmsElementResolver.php:62]: ah okay, then it is okay. just want to make sure, that no data is missing after that change, because that would be breaking

**Comment by MartinKrzykawski** [comment]
Hello @akf-bw,

Thank you very much for your contribution!

Could you please provide more details on what this change specifically addresses? We weren’t aware that manufacturer information is displayed in the product slider.

**Comment by ghost** [comment]
@MartinKrzykawski A product box can use this manufacturer data to, for example, display the manufacturer name directly in the listing.
I don't know if Shopware uses the manufacturer data by default in every box component, but (in most cases at least for custom extensions) this manufacturer data is used / needed.

But the primary goal of this PR change is to make the data returned by the product slider more predictable, so that you always get the same data regardless of whether you select individual products or a product stream.

**Comment by MartinKrzykawski** [comment]
Hello @akf-bw,

thank you for the additional details!

We understand your perspective. Our goal is to avoid loading unnecessary data in the default product slider to prevent potential performance issues. However, we fully agree that extensibility is important, and we recognize the current limitations in the product slider implementation.

We’re actively working on updating the product slider so that the associations provided in the collect function are utilized in the data-fetching process. Since CMS resolvers support event-based extensions, once our changes are merged, you’ll be able to add your chosen associations by using a listener for the cms-slots-data.collect event.

Thank you for your understanding!

Best regards
Martin

**Comment by ghost** [comment]
@MartinKrzykawski But shouldn't we then set the "options.group" and "manufacturer" associations here to "deprecated", for removal with 6.7? Because they are used the same as the associations from the product stream?

**Comment by MartinKrzykawski** [comment]
@akf-bw Yes, I created a ticket to check our CMS resolvers and deprecate unnecessary associations: https://github.com/shopware/shopware/issues/5326

Thank you for bringing this to our attention! :)

**Comment by ghost** [comment]
@MartinKrzykawski Should i implement this deprecation in this PR (you reopen it, i rename) or open a new one?

**Comment by MartinKrzykawski** [comment]
@akf-bw 

If you would like to help us, feel free to work on this and deprecate any unnecessary associations. :) 
Please create a new PR, so you can include the ticket number from the newly created issue.

You will find the CMS resolvers in:
`src/Core/Content/Cms/DataResolver/Element/*`and `src/Core/Content/Product/Cms/*`.

Thanks so much for your help!
